--- a/libarchive/filter_fork_posix.c
+++ b/libarchive/filter_fork_posix.c
@@ -72,6 +72,11 @@
 
 #include "filter_fork.h"
 
+#ifdef TARGET_DARWIN_TVOS
+#undef HAVE_POSIX_SPAWNP
+#undef HAVE_VFORK
+#endif
+
 pid_t
 __archive_create_child(const char *cmd, int *child_stdin, int *child_stdout)
 {
@@ -148,7 +153,11 @@
 #if HAVE_VFORK
 	child = vfork();
 #else
-	child = fork();
+#ifdef TARGET_DARWIN_TVOS
+       child = -1;
+#else
+       child = fork();
+#endif
 #endif
 	if (child == -1)
 		goto stdout_opened;
